packages/mui-utils/src/resolveProps.test.ts

```diff
@@ -44,4 +44,40 @@ describe('resolveProps', () => {
       foo: '',
     });
   });
+
+  it('merge components and componentsProps props', () => {
```

Comment:
Maybe add a test with an object prop not matching this pattern and check that it is not merged

Reply:
Added.